crates/core/src/db/datastore/locking_tx_datastore/mut_tx.rs

```diff
@@ -1250,26 +1244,42 @@ impl MutTxId {
                         // - Insert Row A
                         // This is impossible to recover if `Running 2` elides its insert.
                         tx_table
-                            .delete(tx_blob_store, ptr, |_| ())
+                            .delete(tx_blob_store, tx_row_ptr, |_| ())
                             .expect("Failed to delete a row we just inserted");
 
                         // It's possible that `row` appears in the committed state,
                         // but is marked as deleted.
                         // In this case, undelete it, so it remains in the committed state.
-                        delete_table.remove(&committed_ptr);
+                        delete_table.remove(&commit_ptr);
 
                         // No new row was inserted, but return `committed_ptr`.
                         let blob_store = &self.committed_state_write_lock.blob_store;
                         return Ok(RowRefInsertion::Existed(
                             // SAFETY: `find_same_row` told us that `ptr` refers to a valid row in `commit_table`.
-                            unsafe { commit_table.get_row_ref_unchecked(blob_store, committed_ptr) },
+                            unsafe { commit_table.get_row_ref_unchecked(blob_store, commit_ptr) },
                         ));
                     }
+
+                    // Pacify the borrow checker.
+                    // SAFETY: `ptr` came from `tx_table.insert` just now without any interleaving calls.
```

Comment:
This comment is no longer accurate. `tx_row_ptr` (which is renamed) came from the `tx_row_ref`. There have been intervening calls - `Table::find_same_row` notably - and this comment should justify (or at least assert) that the pointer is not invalidated by those calls.